deepchecks/vision/checks/performance/single_dataset_scalar_performance.py

```diff
@@ -0,0 +1,127 @@
+from deepchecks.vision.metrics_utils.object_detection_precision_recall import \
+    ObjectDetectionAveragePrecision
+import warnings
+
+__all__ = ['SingleDatasetScalarPerformance']
+
+
+class SingleDatasetScalarPerformance(SingleDatasetCheck):
+    """Calculate a performance metric as a scalar for a given model and a given dataset.
+
+    Parameters
+    ----------
+        metric: Metric,  default: None
+        An ignite.Metric object whose score should be used. If None is given, use the default metric.
+        reduce: torch function, default: None
+        The function to reduce the scores tensor into a single scalar. For metrics that return a scalar use None
+        (default).
+    """
+    def __init__(self,
+                 metric: Metric = None,
+                 reduce: t.Callable = None,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.metric = metric
+        self.reduce = reduce
+
+    def initialize_run(self, context: Context, dataset_kind: DatasetKind.TRAIN):
+        """Initialize the metric for the check, and validate task type is relevant."""
+        if self.metric is None:
+            if context.train.task_type == TaskType.CLASSIFICATION:
+                self.metric = Accuracy()
+            elif context.train.task_type == TaskType.OBJECT_DETECTION:
+                self.metric = ObjectDetectionAveragePrecision()
+                if self.reduce is None:
+                    self.reduce = torch.nanmean
+            else:
+                raise DeepchecksValueError('For task types other then classification or object detection, '
+                                           'pass a metric explicitly')
+        self.metric.reset()
+
+    def update(self, context: Context, batch: Batch, dataset_kind: DatasetKind.TRAIN):
+        """Update the metrics by passing the batch to ignite metric update method."""
+        label = batch.labels
+        prediction = batch.predictions
+        self.metric.update((prediction, label))
+
+    def compute(self, context: Context, dataset_kind: DatasetKind.TRAIN) -> CheckResult:
+        """Compute the metric result using the ignite metrics compute method and reduce to a scalar."""
+        metric_result = self.metric.compute()
+        if self.reduce is not None:
+            if type(metric_result) is float:
+                warnings.warn(SyntaxWarning('Metric result is already scalar, skipping reduce operation'))
+                result_value = metric_result
+            else:
+                result_value = float(self.reduce(metric_result))
+        elif type(metric_result) is float:
+            result_value = metric_result
+        else:
+            raise DeepchecksValueError(f'The metric {self.metric.__class__} return a non-scalar value, '
+                                       f'please specify a reduce function or choose a different metric')
+        result_dict = {'score': result_value,
+                       'metric': self.metric.__class__.__name__,
```

Comment:
What is self.metric.__class__.__name__ for example for ObjectDetectionAveragePrecision? Because if it's 'ObjectDetectionAveragePrecision' we cant show it to the user like this. We probably should receive a dict {metric_name: metric_value} for the metric argument if that is the case, so the user / our default values will be the ones deciding the display name.